org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacCompilationUnitResolver.java

```diff
@@ -384,28 +384,32 @@ private void resolveBindings(CompilationUnit unit, int apiLevel) {
 	}
 
 	private void resolveBindings(CompilationUnit unit, Map<String, IBinding> bindingMap, int apiLevel) {
-		if (unit.getPackage() != null) {
-			IPackageBinding pb = unit.getPackage().resolveBinding();
-			if (pb != null) {
-				bindingMap.put(pb.getKey(), pb);
+		try {
+			if (unit.getPackage() != null) {
+				IPackageBinding pb = unit.getPackage().resolveBinding();
+				if (pb != null) {
+					bindingMap.put(pb.getKey(), pb);
+				}
 			}
-		}
-		if (!unit.types().isEmpty()) {
-			List<AbstractTypeDeclaration> types = unit.types();
-			for( int i = 0; i < types.size(); i++ ) {
-				ITypeBinding tb = ((AbstractTypeDeclaration) types.get(i)).resolveBinding();
-				if (tb != null) {
-					bindingMap.put(tb.getKey(), tb);
+			if (!unit.types().isEmpty()) {
+				List<AbstractTypeDeclaration> types = unit.types();
+				for( int i = 0; i < types.size(); i++ ) {
+					ITypeBinding tb = ((AbstractTypeDeclaration) types.get(i)).resolveBinding();
+					if (tb != null) {
+						bindingMap.put(tb.getKey(), tb);
+					}
 				}
 			}
-		}
-		if( apiLevel >= AST.JLS9_INTERNAL) {
-			if (unit.getModule() != null) {
-				IModuleBinding mb = unit.getModule().resolveBinding();
-				if (mb != null) {
-					bindingMap.put(mb.getKey(), mb);
+			if( apiLevel >= AST.JLS9_INTERNAL) {
+				if (unit.getModule() != null) {
+					IModuleBinding mb = unit.getModule().resolveBinding();
+					if (mb != null) {
+						bindingMap.put(mb.getKey(), mb);
+					}
 				}
 			}
+		} catch (Exception e) {
```

Comment:
I don't think we want to silent this exception here. We should instead throw it, and consider catching it in the caller.

Reply:
I agree with you. For this instance i could have handle it much higher level. But it still could be somewhere which is not diagnostic related due this call comes from jdt.ls. But i can check. 

I was also thinking to avoid failures in other scenarios due to syntax errors. It would be really nice if we can check for syntax errors and control throw or not to throw. 

I tried checking the flags of the unit to see if it is marked as malformed, but it was not.